apps/segment/src/modules/webhooks/webhook-activity/webhook-activity-service.ts

```diff
@@ -0,0 +1,75 @@
+import { err, ok, Result } from "neverthrow";
+
+import { BaseError } from "@/errors";
+
+import { IWebhooksActivityClient } from "./webhook-activity-client";
+
+export interface IWebhookActivityService {
+  enableAppWebhooks(): Promise<Result<undefined, unknown>>;
+  getWebhooksIsActive(): Promise<Result<boolean[], unknown>>;
+}
+
+/**
+ * Service responsible for managing app webhooks `isActive` state.
+ */
+export class WebhookActivityService implements IWebhookActivityService {
+  static WebhookActivityServiceWebhooksError = BaseError.subclass(
+    "WebhookActivityServiceWebhooksError",
+  );
+
+  constructor(
+    private ownAppId: string,
+    private client: IWebhooksActivityClient,
+  ) {}
+
+  async getWebhooksIsActive() {
+    const webhooksIds = await this.client.fetchAppWebhooksInformation(this.ownAppId);
+
+    if (webhooksIds.isErr()) {
+      return err(
+        new WebhookActivityService.WebhookActivityServiceWebhooksError(
+          "Error while fetching webhooks IDs",
+          {
+            cause: webhooksIds.error,
+          },
+        ),
+      );
+    }
+
+    return ok(webhooksIds.value.map((webhook) => webhook.isActive));
+  }
+
+  async enableAppWebhooks() {
+    const webhooksResult = await this.client.fetchAppWebhooksInformation(this.ownAppId);
+
+    if (webhooksResult.isErr()) {
+      return err(
+        new WebhookActivityService.WebhookActivityServiceWebhooksError(
+          "Error while fetching webhooks IDs",
+          {
+            cause: webhooksResult.error,
+          },
+        ),
+      );
+    }
+
+    const appWebhooksEnableResult = Result.combine(
+      await Promise.all(
+        webhooksResult.value.map((webhook) => this.client.enableSingleWebhook(webhook.id)),
+      ),
+    );
```

Comment:
suggestion: maybe we can make feature request to Saleor API so that we have a mutation to update multiple webhooks at the same time?

If one of these updates fails we will have partial state when some webhooks are enabled and some are not :/

Reply:
Yes - I will create issue for that 👍🏻